src/libs/nav2_tasks/include/nav2_tasks/task_client.hpp

```diff
@@ -35,7 +35,7 @@ template<class CommandMsg, class ResultMsg>
 class TaskClient
 {
 public:
-  explicit TaskClient(rclcpp::Node * node)
+  explicit TaskClient(rclcpp::Node::SharedPtr node)
```

Comment:
I've been using shared_ptr & in cases like these. Without the reference, we create a temporary shared_ptr for the parameter, then another one for the node_. Then we have to destroy the temporary. By making it a &, the temporary shared_ptr goes away.

It might be possible to make it a const &, ideally, but I'm not sure if you can assign a const shared_ptr to a non-const.